kernels/absolute.cl

```diff
@@ -0,0 +1,18 @@
+__constant sampler_t sampler = CLK_NORMALIZED_COORDS_FALSE | CLK_ADDRESS_CLAMP_TO_EDGE | CLK_FILTER_NEAREST;
+
+__kernel void absolute(
+    IMAGE_src_TYPE  src,
+    IMAGE_dst_TYPE  dst
+)
+{
+  const int x = get_global_id(0);
+  const int y = get_global_id(1);
+  const int z = get_global_id(2);
+
+  float value = READ_IMAGE(src, sampler, POS_src_INSTANCE(x,y,z,0)).x;
+  if ( value < 0 ) {
+    value = -1 * value;
+  }
+
+  WRITE_IMAGE (dst, POS_dst_INSTANCE(x,y,z,0), CONVERT_dst_PIXEL_TYPE(value));
```

Comment:
I continue with my questions, shouldn't we use READ_src_IMAGE, WRITE_dst_IMAGE ? 

```suggestion
  float value = READ_src_IMAGE(src, sampler, POS_src_INSTANCE(x,y,z,0)).x;
  if ( value < 0 ) {
    value = -1 * value;
  }

  WRITE_dst_IMAGE (dst, POS_dst_INSTANCE(x,y,z,0), CONVERT_dst_PIXEL_TYPE(value));
```

Reply:
`READ_IMAGE(src,...` is a short-cut to `READ_src_IMAGE(src,...` and easier to read/write I'd say.

Reply:
https://github.com/clEsperanto/CLIc_prototype/blob/master/clic/preamble.cl#L466

Reply:
no, never, because `READ_IMAGE(xyz, ...` will be replaced by `READ_xyz_IMAGE(xyz, ...` 😎